core/src/main/java/eu/fasten/core/data/ExtendedRevisionJavaCallGraph.java

```diff
@@ -169,4 +173,63 @@ public String getRevisionName() {
         return artifactId + "_" + groupId + "_" + this.version;
     }
 
+    @Override
```

Comment:
Maybe implementing `equals` and `hashCode` in parent class that actually declare the field `nodeCount`, `graph` would make more sense.

Reply:
Thanks for the comment, I agree, In fact, all of the data classes must have equals and hashCode. In this PR I had to implement a java specific one to do the testing but we will have another PR in the future for refactoring the whole data package that needs to address all these.

Reply:
Ok, thanks for the clarification.